v2/internal/controllers/crd_machinelearningservices_test.go

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package controllers_test
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-service-operator/v2/api/compute/v1beta20201201"
+	"github.com/Azure/azure-service-operator/v2/api/keyvault/v1beta20210401preview"
+	machinelearningservices "github.com/Azure/azure-service-operator/v2/api/machinelearningservices/v1beta20210701"
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1beta20201101"
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1beta20200601"
+	storage "github.com/Azure/azure-service-operator/v2/api/storage/v1beta20210401"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+func Test_MachineLearning_Workspaces_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	sa := newStorageAccount(tc, rg)
+
+	tc.CreateResourceAndWait(sa)
+
+	kv := newVault(tc, rg)
+	tc.CreateResourceAndWait(kv)
+
+	// Have to use 'eastus' location here as 'ListKeys' API is unavailable/still broken for 'westus2'
+	workspace := newWorkspace(tc, testcommon.AsOwner(rg), sa, kv, to.StringPtr("eastus"))
+
+	tc.CreateResourcesAndWait(workspace)
+
+	Workspaces_WriteSecrets(tc, workspace)
```

Comment:
Oh don't call this directly. Call it via `tc.RunSubtests`. That's the sequential version.